riscv/src/compiler.rs

```diff
@@ -1428,6 +1433,40 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             ]
         }
 
+        insn if insn.starts_with("lr.w") => {
+            // Very similar to "lw":
+            let (rd, rs, off) = rro(args);
+            assert_eq!(off, 0);
+            // TODO misaligned access should raise misaligned address exceptions
+            only_if_no_write_to_zero_vec(
+                vec![
+                    format!("addr <=X= {rs};"),
+                    format!("{rd} <== mload();"),
```

Comment:
The standard says you have to sign-extend the value, or did I read that wrong?

Reply:
I think that part is only relevant for 64-bits when using `lr.w`. On 32 bits, `lr.w` is already the size of the register.